File: app/src/main/java/com/refugio/pawrescue/ui/publico/PublicRegisterActivity.java
package com.refugio.pawrescue.ui.publico;

import android.content.Intent;
import android.os.Bundle;
import android.util.Patterns;
import android.view.View;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;
import com.refugio.pawrescue.databinding.ActivityPublicRegisterBinding;
import com.refugio.pawrescue.ui.auth.LoginActivity;

public class PublicRegisterActivity extends AppCompatActivity {

    private ActivityPublicRegisterBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityPublicRegisterBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setupUI();
    }

    private void setupUI() {
        binding.btnRegistrar.setOnClickListener(v -> {
            String nombre = binding.etNombre.getText().toString().trim();
            String email = binding.etEmail.getText().toString().trim();
            String password = binding.etPassword.getText().toString().trim();
            String passwordConfirm = binding.etPasswordConfirm.getText().toString().trim();
            String telefono = binding.etTelefono.getText().toString().trim();

            if (validateInputs(nombre, email, password, passwordConfirm, telefono)) {
                register(nombre, email, password, telefono);
            }
        });

        binding.tvYaTienesCuenta.setOnClickListener(v -> {
            startActivity(new Intent(this, LoginActivity.class));
            finish();
        });
    }

    private boolean validateInputs(
            String nombre,
            String email,
            String password,
            String passwordConfirm,
            String telefono
    ) {
        boolean isValid = true;

        if (nombre.isEmpty()) {
            binding.tilNombre.setError("El nombre es requerido");
            isValid = false;
        } else {
            binding.tilNombre.setError(null);
        }

        if (email.isEmpty()) {
            binding.tilEmail.setError("El email es requerido");
            isValid = false;
        } else if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
            binding.tilEmail.setError("Email inválido");
            isValid = false;
        } else {
            binding.tilEmail.setError(null);
        }

        if (password.isEmpty()) {
            binding.tilPassword.setError("La contraseña es requerida");
            isValid = false;
        } else if (password.length() < 6) {
            binding.tilPassword.setError("Mínimo 6 caracteres");
            isValid = false;
        } else {
            binding.tilPassword.setError(null);
        }

        if (!passwordConfirm.equals(password)) {
            binding.tilPasswordConfirm.setError("Las contraseñas no coinciden");
            isValid = false;
        } else {
            binding.tilPasswordConfirm.setError(null);
        }

        if (telefono.isEmpty()) {
            binding.tilTelefono.setError("El teléfono es requerido");
            isValid = false;
        } else {
            binding.tilTelefono.setError(null);
        }

        return isValid;
    }

    private void register(String nombre, String email, String password, String telefono) {
        showLoading(true);

        // lógica de registro llamar a Firebase

        /*
        FirebaseAuth auth = FirebaseAuth.getInstance();
        auth.createUserWithEmailAndPassword(email, password)
            .addOnCompleteListener(this, task -> {
                if (task.isSuccessful()) {
                    // Guardar datos adicionales en Firestore o Realtime Database
                    saveUserData(nombre, email, telefono);
                } else {
                    showLoading(false);
                    Toast.makeText(this,
                        "Error: " + task.getException().getMessage(),
                        Toast.LENGTH_LONG).show();
                }
            });
        */

        new android.os.Handler().postDelayed(() -> {
            showLoading(false);
            Toast.makeText(
                    this,
                    "✅ Registro exitoso. ¡Bienvenido!",
                    Toast.LENGTH_LONG
            ).show();
            navigateToPublicMain();
        }, 2000);
    }

    private void showLoading(boolean show) {
        binding.progressBar.setVisibility(show ? View.VISIBLE : View.GONE);
        binding.btnRegistrar.setEnabled(!show);
    }

    private void navigateToPublicMain() {
        Intent intent = new Intent(this, PublicMainActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/com/refugio/pawrescue/ui/theme/public_user/PublicRegisterActivity.kt
package com.refugio.pawrescue.ui.theme.public_user

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import com.refugio.pawrescue.databinding.ActivityPublicRegisterBinding
import com.refugio.pawrescue.ui.theme.auth.LoginActivity

class PublicRegisterActivity : AppCompatActivity() {

    private lateinit var binding: ActivityPublicRegisterBinding
    private val viewModel: PublicRegisterViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityPublicRegisterBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setupUI()
        observeViewModel()
    }

    private fun setupUI() {
        binding.btnRegistrar.setOnClickListener {
            val nombre = binding.etNombre.text.toString().trim()
            val email = binding.etEmail.text.toString().trim()
            val password = binding.etPassword.text.toString().trim()
            val passwordConfirm = binding.etPasswordConfirm.text.toString().trim()
            val telefono = binding.etTelefono.text.toString().trim()

            if (validateInputs(nombre, email, password, passwordConfirm, telefono)) {
                viewModel.register(nombre, email, password, telefono)
            }
        }

        binding.tvYaTienesCuenta.setOnClickListener {
            startActivity(Intent(this, LoginActivity::class.java))
            finish()
        }
    }

    private fun validateInputs(
        nombre: String,
        email: String,
        password: String,
        passwordConfirm: String,
        telefono: String
    ): Boolean {
        var isValid = true

        if (nombre.isEmpty()) {
            binding.tilNombre.error = "El nombre es requerido"
            isValid = false
        } else {
            binding.tilNombre.error = null
        }

        if (email.isEmpty()) {
            binding.tilEmail.error = "El email es requerido"
            isValid = false
        } else if (!android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
            binding.tilEmail.error = "Email inválido"
            isValid = false
        } else {
            binding.tilEmail.error = null
        }

        if (password.isEmpty()) {
            binding.tilPassword.error = "La contraseña es requerida"
            isValid = false
        } else if (password.length < 6) {
            binding.tilPassword.error = "Mínimo 6 caracteres"
            isValid = false
        } else {
            binding.tilPassword.error = null
        }

        if (passwordConfirm != password) {
            binding.tilPasswordConfirm.error = "Las contraseñas no coinciden"
            isValid = false
        } else {
            binding.tilPasswordConfirm.error = null
        }

        if (telefono.isEmpty()) {
            binding.tilTelefono.error = "El teléfono es requerido"
            isValid = false
        } else {
            binding.tilTelefono.error = null
        }

        return isValid
    }

    private fun observeViewModel() {
        viewModel.registerState.observe(this) { state ->
            when (state) {
                is RegisterState.Loading -> showLoading(true)
                is RegisterState.Success -> {
                    showLoading(false)
                    Toast.makeText(
                        this,
                        "Registro exitoso. Bienvenido!",
                        Toast.LENGTH_LONG
                    ).show()
                    navigateToPublicMain()
                }
                is RegisterState.Error -> {
                    showLoading(false)
                    Toast.makeText(this, state.message, Toast.LENGTH_LONG).show()
                }
                else -> showLoading(false)
            }
        }
    }

    private fun showLoading(show: Boolean) {
        binding.progressBar.visibility = if (show) View.VISIBLE else View.GONE
        binding.btnRegistrar.isEnabled = !show
    }

    private fun navigateToPublicMain() {
        val intent = Intent(this, PublicMainActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        startActivity(intent)
        finish()
    }
}